### ISSUE TITLE
dt_install_on_the_fly() messes up -ado dir-

### ISSUE THREAD

**Issue by bquistorff** [gvegayon/devtools#1]
Repeatedly calling dt_install_on_the_fly() makes duplicate entries in the output of -ado dir- (which reads from the stata.trk file). This is really a problem with -net install, force-, but maybe something sould be done here. Do you you want to uninstall be installing? Or are you worried the install won't succeed and therefore leave the the ado folder worse?

**Comment by gvegayon** [comment]
Can you give me an example?? Thanks

**Comment by bquistorff** [comment]
After running parallel/ado/compile.do lots times during the course of development, now when I run -ado dir-, of the 76 entries listed, 43 are -parallel-. It's annoying to look through, but not something very bad.

**Comment by bquistorff** [comment]
After a bit more investigating it looks like this can actually cause a problem with -ado uninstall- if one has installed from different sources over time. I wrote up the issue more completely [here](http://www.statalist.org/forums/forum/general-stata-discussion/general/719183-dealing-with-duplicate-entries-in-stata-trk).

**Comment by gvegayon** [comment]
Yeah, I can't think on a clean solution right now, other than mess with the stata.trk file (which I don't know if it worths the time). Sounds like a technical wish for some Stata conference!
